tests/componenttests/server/tests/mediaKeys/SessionReadyForDecryptionTest.cpp

```diff
@@ -0,0 +1,264 @@
+    EXPECT_CALL(m_ocdmSessionMock, cleanDecryptContext()).WillOnce(Return(MediaKeyErrorStatus::OK));
+}
+void SessionReadyForDecryptionTest::closeKeySessionNetflix()
+{
+    auto request{createCloseKeySessionRequest(m_mediaKeysHandle, m_mediaKeySessionId)};
+
+    ConfigureAction<CloseKeySession>(m_clientStub)
+        .send(request)
+        .expectSuccess()
+        .matchResponse([&](const firebolt::rialto::CloseKeySessionResponse &resp)
+                       { EXPECT_EQ(resp.error_status(), ProtoMediaKeyErrorStatus::OK); });
+}
+
+void SessionReadyForDecryptionTest::destroyMediaKeysRequest()
+{
+    auto request{createDestroyMediaKeysRequest(m_mediaKeysHandle)};
+
+    ConfigureAction<DestroyMediaKeys>(m_clientStub)
+        .send(request)
+        .expectSuccess()
+        .matchResponse([&](const firebolt::rialto::DestroyMediaKeysResponse &resp) {});
+}
+
+/*
+ * Component Test: Session initalised and ready for decryption for widevine
+ * Test Objective:
+ *  Test the license generation and updating of a key session for widevine media
+ *  key system.
+ *
+ * Sequence Diagrams:
+ *  Create MKS - Cobalt/OCDM, Update MKS - Cobalt/OCDM, "Destroy" MKS - Cobalt/OCDM
+ *   - https://wiki.rdkcentral.com/display/ASP/Rialto+Media+Key+Session+Management+Design
+ *
+ * Test Setup:
+ *  Language: C++
+ *  Testing Framework: Google Test
+ *  Components: MediaKeys
+ *
+ * Test Initialize:
+ *   RialtoServerComponentTest::RialtoServerComponentTest() will set up wrappers and
+ *      starts rialtoServer running in its own thread
+ *   send a CreateMediaKeys message to rialtoServer
+ *   expect a "createSession" call (to OCDM mock)
+ *   send a CreateKeySession message to rialtoServer
+ *
+ *  Step 1: Generate license request
+ *   Server notifies the client of license request.
+ *   Expect that the license request notification is propagated to the client.
+ *
+ *  Step 2: Update session
+ *   updateSession with a key.
+ *   Expect that updateSession is propagated to the server.
+ *   Api call returns with success.
+ *   Server notifys the client of key statuses changed.
+ *   Expect that the key statuses changed notification is propagated to the client.
+ *
+ *  Step 3: Close session
+ *   closeSession.
+ *   Expect that closeSession is propagated to the server.
+ *   Api call returns with success.
+ *
+ *  Step 4: Destroy media keys
+ *   Destroy instance of MediaKeys.
+ *   Expect that media keys is destroyed on the server.
+ *
+ * Test Teardown:
+ *  Server is terminated.
+ *
+ * Expected Results:
+ *  For a widevine key system, the application can create new sessions, generate license
+ *  requests and update keys ready for decryption of content.
+ *
+ * Code:
+ */
+TEST_F(SessionReadyForDecryptionTest, shouldUpdateWidevine)
+{
+    createMediaKeysWidevine();
+    ocdmSessionWillBeCreated();
+    createKeySession();
+
+    // Step 1: Generate license request
+    licenseRenew();
```

Comment:
I would expect a call to generate request with a license request callback rather than renewal here.

Additionally, in the tests in client side the generate request step is subtlety different for both widevine and netflix, for Netflix/playready we expect a license request notification before the generateRequest API call returns, for widevine generateRequest completes and then we receive a notification. This can be seen in the Create Cobalt and Create Netflix sequence diagrams.